### PR TITLE
Implemented Encrypted Secrets for GitOps Style of Spinnaker Implement…

### PR DESCRIPTION
**Pull request by vkvamsiopsmx**

### CODE REVIEW

File: charts/oes/charts/spinnaker/templates/configmap/halyard-init-script.yaml [hunk 1]
```diff
@@ -47,7 +47,9 @@ data:
     {{- else }}
     #!/bin/bash -x
     rm -rf /tmp/spinnaker/.hal
-    git clone $GIT_CLONE_PARAM /tmp/spinnaker/.hal
+    git clone $GIT_CLONE_PARAM /tmp/spinnaker/test
+    cp -pr /tmp/spinnaker/test/{{ .Values.gitopsHalyard.repo.halConfigPath }} /tmp/spinnaker/.hal
```
Comment (abhinaybyrisetty) [inline charts/oes/charts/spinnaker/templates/configmap/halyard-init-script.yaml:51]: Why do we need this intermediate step here?

File: charts/oes/templates/secrets/opsmx-gitops-secret.yaml [hunk 2]
```diff
@@ -3,10 +3,10 @@
 apiVersion: v1
 stringData:
   # Repo uri to fetch halyard configuration
-  gitcloneparam: https://{{ .Values.spinnaker.gitopsHalyard.repo.username }}:{{ .Values.spinnaker.gitopsHalyard.repo.token }}@github.com/{{ .Values.spinnaker.gitopsHalyard.repo.organization }}/{{ .Values.spinnaker.gitopsHalyard.repo.halRepository }}.git
+  gitcloneparam: https://{{ .Values.spinnaker.gitopsHalyard.repo.username }}:{{ .Values.spinnaker.gitopsHalyard.repo.token }}@github.com/{{ .Values.spinnaker.gitopsHalyard.repo.organization }}/{{ .Values.spinnaker.gitopsHalyard.repo.Repository }}.git
```
Comment (abhinaybyrisetty) [inline charts/oes/templates/secrets/opsmx-gitops-secret.yaml:6]: Let's use repository in lower case just to maintain the standard.

File: charts/oes/values.yaml [hunk 3]
```diff
@@ -977,8 +977,9 @@ spinnaker:
     repo:
       type: git # git, S3, vault, only git support at this point
       organization: OpsMx
-      halRepository: gitops-haldir # Please provide the repo URL of the GitOps Halyard Directory.
-      dynamicAccRepository: standard-dynamic-account # Please provide the repo URL of the GitOps Dynamic Accounts Directory.
+      Repository: standard-gitops-repo # Please provide the repo URL of the GitOps Halyard Directory.
```
Comment (abhinaybyrisetty) [inline charts/oes/values.yaml:980]: this should be 'repository' in smaller case and the let's give the value in the form of a url, not as invalid keyword. Let's give our sample repos that we created by default in here.

https://github.com/OpsMx/gitops-hal-sample.git
https://github.com/OpsMx/gitops-accounts-sample.git

**Review by abhinaybyrisetty — CHANGES_REQUESTED** [review]
Please address the minor changes, if the script in configmap is tested already, I don't have any comments.

File: charts/oes/charts/spinnaker/templates/configmap/halyard-init-script.yaml (hunk 1 above)
Comment (vkvamsiopsmx) [reply charts/oes/charts/spinnaker/templates/configmap/halyard-init-script.yaml:51]: We are implementing the folder structure here,to handle all the three gitops repo, dynamic accounts, pipeline promotion in one single git repo.

File: charts/oes/values.yaml (hunk 3 above)
Comment (vkvamsiopsmx) [reply charts/oes/values.yaml]: We need to use this url in multiple places and in some cases we need it to enter along with user name and token, so we can not make in the form of url. The same is already present in our existing version, its just updated now with one parameter.

File: charts/oes/templates/secrets/opsmx-gitops-secret.yaml (hunk 2 above)
Comment (vkvamsiopsmx) [reply charts/oes/templates/secrets/opsmx-gitops-secret.yaml]: Updated it.

File: charts/oes/values.yaml (hunk 3 above)
Comment (abhinaybyrisetty) [reply charts/oes/values.yaml]: Having it as url gives user an indication of how url should be. Otherwise, let's add detailed comments please

Comment (abhinaybyrisetty) [reply charts/oes/values.yaml]: Reason why I'm specifically stressing on specifying sample url is, if user doesn't keep ".git" suffix at the end, it won't clone.

Comment (vkvamsiopsmx) [reply charts/oes/values.yaml]: .git part is defined in the templates.

Comment (abhinaybyrisetty) [reply charts/oes/values.yaml]: Let's go with eith "null" or a valid input that's indicative

Comment (vkvamsiopsmx) [reply charts/oes/values.yaml]: @abhinaybyrisetty  have added new commit with the updated comments, please validate

**Review by abhinaybyrisetty — APPROVED** [review]
LGTM

**Pull request merged**